src/hotspot/share/opto/phaseX.cpp

```diff
@@ -62,6 +62,31 @@ NodeHash::NodeHash(Arena *arena, uint est_max_size) :
   memset(_table,0,sizeof(Node*)*_max);
 }
 
+//-----------------------------------------------------------------------------
+
+bool NodeHash::check_for_collision(const Node* n, const Node* k) const {
+  // For commutative operations with same controlling edge
+  // perform order agnostic input edge comparison to promote
+  // node sharing.
+  uint req = n->req();
+  if (n->is_commutative_vector_operation()) {
+    assert(req == 3, "");
+    assert(k->is_commutative_vector_operation(), "");
+    if ((k->in(0) != n->in(0)) ||
+        ((k->in(1) != n->in(1) || k->in(2) != n->in(2)) &&
+         (k->in(1) != n->in(2) || k->in(2) != n->in(1)))) {
+      return true;
+    }
+  } else {
+    for(uint i=0; i<req; i++) {
+      if(n->in(i) != k->in(i)) { // Different inputs?
+        return true;
+      }
+    }
```

Comment:
```suggestion
    for (uint i = 0; i < req; i++) {
      if (n->in(i) != k->in(i)) { // Different inputs?
        return true;
      }
    }
```
I know you just copied it, but we should fix formatting if we are touching the code anyway ;)